From 43630e7aa7ede6f0661b728915a00ac6ddc645dd Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 14 Oct 2012 10:27:38 +0200
Subject: [PATCH 4/4] Add the _cached and _writecombine function APIs

Those functions are used to allocate buffers with writecombine attribute on CPU bus, or completely as a cached area.
BMM driver takes care of page table manipulation.

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 xbmc/cores/dvdplayer/DVDCodecs/Video/DllVMETA.h |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DllVMETA.h b/xbmc/cores/dvdplayer/DVDCodecs/Video/DllVMETA.h
index bbead6d..5553f5e 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DllVMETA.h
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DllVMETA.h
@@ -70,6 +70,8 @@ public:
   virtual IppCodecStatus DecodeSendCmd_Vmeta(int cmd, void *pInParam, void *pOutParam, void *pSrcDecoderState) = 0;
   virtual IppCodecStatus DecodeFrame_Vmeta(IppVmetaDecInfo *pDecInfo, void *pSrcDstDecoderState) = 0;
   virtual void *vdec_os_api_dma_alloc(UNSG32 size, UNSG32 align, UNSG32 *pPhysical) = 0;
+  virtual void *vdec_os_api_dma_alloc_cached(UNSG32 size, UNSG32 align, UNSG32 *pPhysical) = 0;
+  virtual void *vdec_os_api_dma_alloc_writecombine(UNSG32 size, UNSG32 align, UNSG32 *pPhysical) = 0;
   virtual void vdec_os_api_dma_free(void *ptr) = 0;
   virtual SIGN32 vdec_os_api_suspend_check(void) = 0;
   virtual void vdec_os_api_suspend_ready(void) = 0;
@@ -86,6 +88,8 @@ class DllLibVMETA : public DllDynamic, DllLibVMETAInterface
   DEFINE_METHOD4(IppCodecStatus, DecodeSendCmd_Vmeta,     (int p1, void *p2, void *p3, void *p4))
   DEFINE_METHOD2(IppCodecStatus, DecodeFrame_Vmeta,       (IppVmetaDecInfo *p1, void *p2))
   DEFINE_METHOD3(void *,         vdec_os_api_dma_alloc,   (UNSG32 p1, UNSG32 p2, UNSG32 *p3))
+  DEFINE_METHOD3(void *,         vdec_os_api_dma_alloc_cached,   (UNSG32 p1, UNSG32 p2, UNSG32 *p3))
+  DEFINE_METHOD3(void *,         vdec_os_api_dma_alloc_writecombine,   (UNSG32 p1, UNSG32 p2, UNSG32 *p3))
   DEFINE_METHOD1(void,           vdec_os_api_dma_free,    (void *p1))
   DEFINE_METHOD0(SIGN32,         vdec_os_api_suspend_check)
   DEFINE_METHOD0(void,           vdec_os_api_suspend_ready)
@@ -98,6 +102,8 @@ class DllLibVMETA : public DllDynamic, DllLibVMETAInterface
     RESOLVE_METHOD(DecodeSendCmd_Vmeta)
     RESOLVE_METHOD(DecodeFrame_Vmeta)
     RESOLVE_METHOD(vdec_os_api_dma_alloc)
+    RESOLVE_METHOD(vdec_os_api_dma_alloc_cached)
+    RESOLVE_METHOD(vdec_os_api_dma_alloc_writecombine)
     RESOLVE_METHOD(vdec_os_api_dma_free)
     RESOLVE_METHOD(vdec_os_api_suspend_check)
     RESOLVE_METHOD(vdec_os_api_suspend_ready)
-- 
1.7.9.5

